process/message.go

```diff
@@ -192,12 +194,21 @@ func (prevote *Prevote) BlockHash() id.Hash {
 	return prevote.blockHash
 }
 
+func (prevote *Prevote) NilReasons() NilReasons {
+	return prevote.nilReasons
+}
+
 func (prevote *Prevote) Type() MessageType {
 	return PrevoteMessageType
 }
 
 func (prevote *Prevote) String() string {
-	return fmt.Sprintf("Prevote(Height=%v,Round=%v,BlockHash=%v)", prevote.Height(), prevote.Round(), prevote.BlockHash())
+	nilReasonsBytes, err := prevote.NilReasons().MarshalBinary()
+	if err != nil {
+		return fmt.Sprintf("Prevote(Height=%v,Round=%v,BlockHash=%v)", prevote.Height(), prevote.Round(), prevote.BlockHash())
+	}
+	nilReasonsHash := id.Hash(sha256.Sum256(nilReasonsBytes))
```

Comment:
This can be done in a more re-usable way by having the `NilReasons` type implement the `String()` method.